src/database/entities/api.key.entity.ts

```diff
@@ -56,11 +72,21 @@ export class APIKeyEntity {
 		return this.expiresAt < new Date();
 	}
 
-	constructor(apiKeyId: string, apiKey: string, expiresAt: Date, customer: CustomerEntity, user: UserEntity) {
-		this.apiKeyId = apiKeyId;
+	constructor(
+		apiKeyHash: string,
+		apiKey: string,
+		name: string,
+		expiresAt: Date,
+		customer: CustomerEntity,
+		user: UserEntity,
+		revoked = false
+	) {
+		this.apiKeyHash = apiKeyHash;
 		this.apiKey = apiKey;
+		this.name = name;
```

Comment:
can we also add a `description` field, maybe an optional `url` as well.  It was required in the UI

Reply:
API key just needs a name, doesn't need a description. Organisation/customer might have a description, which is addressed in #522